src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -263,6 +263,18 @@ internal static ulong ToUInt64(object value)
         #endregion
 
         #region Public Static Methods
+        public static string? GetName<TEnum>(TEnum value) where TEnum : struct, Enum
+            => GetName(typeof(TEnum), value);
+
+        public static string[] GetNames<TEnum>() where TEnum : struct, Enum
+            => GetNames(typeof(TEnum));
+
+        public static TEnum[] GetValues<TEnum>() where TEnum : struct, Enum
+            => (TEnum[])GetValues(typeof(TEnum));
+
+        public static bool IsDefined<TEnum>(TEnum value) where TEnum : struct, Enum
+            => IsDefined(typeof(TEnum), value);
```

Comment:
What is the benefit of this method if it's just going to box the argument anyway?

Reply:
This can be certainly implemented more efficiently. It just needs to do this:

```
https://github.com/dotnet/runtime/blob/25ea6c44cf3140cc9bfbaaff7f01ad2a27414f8e/src/libraries/System.Private.CoreLib/src/System/RuntimeType.cs#L267-L270
```

Reply:
This is true. I will investigate this tomorrow. Just rebasing now to make sure it all builds

Reply:
@stephentoub One benefit would be that the value and type cannot be accidentally mismatched.